wallet-next/src/storage.rs

```diff
@@ -1,30 +1,84 @@
+use std::path::PathBuf;
+
 use penumbra_chain::params::ChainParams;
 use penumbra_crypto::{
     merkle::{NoteCommitmentTree, Tree},
     note::Commitment,
     FieldExt,
 };
 use penumbra_proto::{crypto::FullViewingKey, Message, Protobuf};
-use sqlx::{query, Pool, Sqlite};
+use sqlx::{migrate::MigrateDatabase, query, Pool, Sqlite};
 
 use crate::sync::ScanResult;
 
+const MAX_MERKLE_CHECKPOINTS_CLIENT: usize = 10;
+
 #[derive(Clone)]
 pub struct Storage {
-    pool: Pool<Sqlite>,
+    storage_path: PathBuf,
```

Comment:
I think we probably want to keep the `Pool` on `Storage`, so we don't have to be constantly connecting to it.